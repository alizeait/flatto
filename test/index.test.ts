const mock = require("../mock");
const { flatto } = require("../src");

describe("flatto", () => {
  test("should flatten correctly", () => {
    expect(flatto(mock)).toMatchSnapshot();
  });
  test("should add custom seperator", () => {
    const data = {
      key1: { a: 2 },
      key2: { b: 3 },
    };
    expect(flatto(data, "-")).toEqual({ "key1-a": 2, "key2-b": 3 });
    expect(flatto(mock, "~")).toMatchSnapshot();
  });
  test("should keep an empty object or array", () => {
    const data = {
      key1: { a: [] },
      key2: { b: {} },
    };
    expect(flatto(data)).toEqual({ "key1.a": [], "key2.b": {} });
  });
  test("should keep falsy values", () => {
    const data = {
      key1: { a: false },
      key2: { b: null },
      key3: { c: 0 },
      key4: { d: { e: "" } },
      key5: { f: { g: undefined } },
      key6: { h: { i: NaN } },
    };
    expect(flatto(data)).toEqual({
      "key1.a": false,
      "key2.b": null,
      "key3.c": 0,
      "key4.d.e": "",
      "key5.f.g": undefined,
      "key6.h.i": NaN,
    });
  });
});
