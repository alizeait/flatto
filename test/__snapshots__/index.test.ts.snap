// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flatto should add custom seperator 1`] = `
Object {
  "key1~keyA": "valueI",
  "key2~keyB": "valueII",
  "key3~a~b~c": 2,
  "key4": Array [],
  "key5~a~0": "value1",
  "key5~a~1": "value2",
  "key5~a~2~key1~keyA": "valueI",
  "key5~a~2~key1~keyB~0": 1,
  "key5~a~2~key1~keyB~1": 2,
  "key5~a~2~key1~keyB~2": 3,
  "key5~a~2~key1~keyB~3": 4,
  "key5~b": null,
  "key5~c": undefined,
  "key6": Object {},
}
`;

exports[`flatto should flatten correctly 1`] = `
Object {
  "key1.keyA": "valueI",
  "key2.keyB": "valueII",
  "key3.a.b.c": 2,
  "key4": Array [],
  "key5.a.0": "value1",
  "key5.a.1": "value2",
  "key5.a.2.key1.keyA": "valueI",
  "key5.a.2.key1.keyB.0": 1,
  "key5.a.2.key1.keyB.1": 2,
  "key5.a.2.key1.keyB.2": 3,
  "key5.a.2.key1.keyB.3": 4,
  "key5.b": null,
  "key5.c": undefined,
  "key6": Object {},
}
`;
